src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,169 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidUserDataException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private final StorageDao storageDao = new StorageDaoImpl();
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @Test
+    public void register_existingLogin_notOk() {
+        User user = new User();
+        user.setLogin("TryMyCustomLogin");
+        user.setPassword("1234567");
+        user.setAge(20);
+
+        storageDao.add(user);
+
+        User newUser = new User();
+        newUser.setLogin("TryMyCustomLogin");
+        newUser.setPassword("12345678");
+        newUser.setAge(21);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(newUser);
+        });
+
+    }
+
+    @Test
+    public void login_length_notOk() {
```

Comment:
The test method name `login_length_notOk` should be more descriptive. Consider renaming it to `login_length_tooShort_notOk` to clearly indicate the specific validation being tested.